### DISCUSSION TITLE
DenseSAGEConv vs DenseGraphConv

### DISCUSSION THREAD

**Discussion by smorad** [pyg-team/pytorch_geometric#2502] [Q&A] [answered]
I'm having trouble seeing how the implementation of `DenseSAGEConv` is different from `DenseGraphConv`. It seems like both do `out = adj @ x`, then push the nodes and neighbours through distinct FC layers `lin_root(x)` and `lin_rel(out)`. Isn't `SAGE` supposed to do skip connections and generally be different than `GraphConv`?

**Comment by rusty1s** [comment, marked as answer]
Yes, they are both pretty similar. One difference is the `normalize=True` flag. Furthermore, `DenseGraphConv` has support for different aggregations (such as `add`), while `DenseSAGEConv` always performs (unweighted) mean-aggregation.
